### ISSUE TITLE
docker in docker jobs are eating the cache

### ISSUE THREAD

**Issue by BenTheElder** [kubernetes/test-infra#5617]
Discovered this looking at a strange log reported to me by mtaufen, I did some more digging into how docker persists things and have some changes coming to clean leftover data and fix this up.

/kind bug
/area images
/priority important-soon
/assign

**Comment by BenTheElder** [comment]
FYI @krzyzacy as on-call if you see an error about failing to great the git index just retest the PR.
ssh to the node and `rm -rf /mnt/disks/ssd0/docker-graph`. 

 have something cleaner coming that will let us still cache the pulled images, but the disk should take a while to fill again though so that's enough to fix things temporarily.

This also only affects the 8 nodes we currently have that are large enough for the cross jobs.

**Comment by BenTheElder** [comment]
since you can only see what's going on in the pod logs:
https://pastebin.com/4nphpW3C
> Total reclaimed space: 279.3GB

The kubernetes build system leaves a ton of temporary images and volumes around, cleaning them at the beginning of each job while keeping the huge pulled images:
```
+ docker images --all
REPOSITORY                                         TAG                 IMAGE ID            CREATED             SIZE
k8s.gcr.io/kube-cross                              v1.9.2-1            ba98259b52f7        3 days ago          1.74GB
gcr.io/google_containers/kube-cross                v1.9.2-1            ba98259b52f7        3 days ago          1.74GB
gcr.io/google-containers/debian-iptables-s390x     v9                  ffc2b6c061f2        12 days ago         57.8MB
gcr.io/google-containers/debian-iptables-ppc64le   v9                  8f8013a5f6e5        12 days ago         90.9MB
gcr.io/google-containers/debian-iptables-arm64     v9                  a06f057a1e20        12 days ago         51.1MB
gcr.io/google-containers/debian-iptables-arm       v9                  39f928196727        12 days ago         42.6MB
gcr.io/google-containers/debian-iptables-amd64     v9                  58c9f7d47efb        12 days ago         45.3MB
busybox                                            latest              6ad733544a63        2 weeks ago         1.13MB
arm64v8/busybox                                    latest              011b08de1fc7        2 weeks ago         1.26MB
ppc64le/busybox                                    latest              f4279da41337        2 weeks ago         4.81MB
arm32v7/busybox                                    latest              accea84ede35        2 weeks ago         891kB
s390x/busybox                                      latest              36e959045497        2 weeks ago         4.46MB
```
is pretty easy though, I'll have a fix out shortly.

**Comment by fejta** [comment]
@q-lee can your docker in docker solution help?

**Comment by BenTheElder** [comment]
Fixed in #5619
